mlflow/langchain/__init__.py

```diff
@@ -266,30 +267,31 @@ def load_retriever(persist_directory):
             True, the model must implement `stream` method. If None, streamable is
             set to True if the model implements `stream` method. Default to `None`.
     """
-    import langchain
-    from langchain.schema import BaseRetriever
+    with tempfile.TemporaryDirectory() as temp_dir:
+        import langchain
+        from langchain.schema import BaseRetriever
 
-    lc_model_or_path = _validate_and_prepare_lc_model_or_path(lc_model, loader_fn)
+        lc_model_or_path = _validate_and_prepare_lc_model_or_path(lc_model, loader_fn, temp_dir)
 
-    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+        _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
 
-    path = os.path.abspath(path)
-    _validate_and_prepare_target_save_path(path)
+        path = os.path.abspath(path)
+        _validate_and_prepare_target_save_path(path)
 
-    if isinstance(model_config, str):
-        model_config = _validate_and_get_model_config_from_file(model_config)
+        if isinstance(model_config, str):
+            model_config = _validate_and_get_model_config_from_file(model_config)
 
-    model_code_path = None
-    if isinstance(lc_model_or_path, str):
-        # The LangChain model is defined as Python code located in the file at the path
-        # specified by `lc_model`. Verify that the path exists and, if so, copy it to the
-        # model directory along with any other specified code modules
-        model_code_path = lc_model_or_path
+        model_code_path = None
+        if isinstance(lc_model_or_path, str):
+            # The LangChain model is defined as Python code located in the file at the path
```

Comment:
Can we move all the call chain inside here? for `str` you don't need `_validate_and_prepare_lc_model_or_path` right? you can directly call `_validate_and_get_model_code_path` here, then we can scope down the usage of temp_dir